bioptim/dynamics/configure_problem.py

```diff
@@ -113,18 +124,30 @@ def torque_driven(ocp, nlp, with_contact: bool = False, fatigue: FatigueList = N
         ConfigureProblem.configure_qdot(nlp, True, False)
         ConfigureProblem.configure_tau(nlp, False, True, fatigue)
 
+        if implicit_dynamics:
+            ConfigureProblem.configure_qddot(nlp, False, True)
+            ocp.implicit_constraints.add(ImplicitConstraintFcn.QDDOT, node=Node.ALL_SHOOTING)
+
+        if implicit_soft_contacts:
+            ConfigureProblem.configure_soft_contact_forces(nlp, False, True)
+
         if nlp.dynamics_type.dynamic_function:
             ConfigureProblem.configure_dynamics_function(ocp, nlp, DynamicsFunctions.custom)
         else:
             ConfigureProblem.configure_dynamics_function(
-                ocp, nlp, DynamicsFunctions.torque_driven, with_contact=with_contact, fatigue=fatigue
-            )
+                ocp, nlp, DynamicsFunctions.torque_driven, with_contact=with_contact, fatigue=fatigue,
+                implicit_dynamics=implicit_dynamics)
 
         if with_contact:
             ConfigureProblem.configure_contact_function(ocp, nlp, DynamicsFunctions.forces_from_torque_driven)
 
+        ConfigureProblem.configure_soft_contact_function(ocp, nlp)
+        if implicit_soft_contacts:
+            ocp.implicit_constraints.add(ImplicitConstraintFcn.SOFT_CONTACT_FORCES, node=Node.ALL_SHOOTING)
```

Comment:
When you merge, do not forget to add the IMPLICIT flag

Reply:
flaged !